Cargo.toml

```diff
@@ -40,6 +40,8 @@ quickcheck = { optional = true, version = "0.8", default-features = false }
 serde = { version = "1.0", optional = true }
 serde_derive = { version = "1.0", optional = true }
 rayon = { version = "1.5.3", optional = true }
+dot-parser = { version = "0.3", optional = true}
```

Comment:
I'm sorry, I don't have full bandwidth to review well, but I hope others do.

dot-parser already optionally depends on petgraph. How does it all interact, when petgraph optionally depends on dot-parser?

Reply:
Hi bluss, 

dot-parser depends on petgraph for a single thing: to export parsed graphs as petgraph's `Graph`. (see here : https://codeberg.org/bromind/dot-parser/src/branch/master/src/canonical.rs#L156 ).

Basically, this is a poor attempt to do what I intend to do (in a cleaner way) with this PR: to get petgraphs directly from Dot files. However, implementing that in `dot-parser` is intrinsically bad (imho), since it wouldn't be very flexible in terms of petgraph's version (e.g. as of today, `dot-parser` exports `Graph`s from petgraph 0.6.5, which causes issues if the user wants/needs to use an other version). 

Eventually, if this PR gets accepted, I'll removed the dependency. 

From a more practical aspect, the current status is not so much of a problem, since the PR does not uses the `petgraph` feature of dot-parser, so petgraph should not even appear down the dependency tree/there is not cyclic dependency issue.